src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -162,15 +183,18 @@ public bool TryGetTelemetryId(out Guid telemetryId)
 
                 if (_workspaceStatusService != null)
                 {
-                    // TODO: right now, LightBulb uses IVsThreadedWaitDialog directly rather than new
-                    //       IUIThreadOperationContext. we need to talk to editor team whether we want to
-                    //       update API to pass in OperationContext like any new API, or we use 
-                    //       IWaitIndicator abstraction (which is a thin wrapper on top of IVsThreadedWaitDialog) directly
-                    //       for now, we use the one LB created before calling us. meaning we don't update
-                    //       text on the wait dialog window
-                    // 
-                    //       this also needs to run under threading context otherwise, we can deadlock on VS
-                    ThreadingContext.JoinableTaskFactory.Run(() => _workspaceStatusService.WaitUntilFullyLoadedAsync(cancellationToken));
+                    IDisposable? operationScope = null;
+                    try
```

Comment:
simpler just as:

```c#
if (addOperationScope != null)
{
     using ...
}
```

Then you don't have to deal with null.  (note: i thought `using` was null tolerant anyways).

--

You can also simplify by not passing a null delegate along.  Just have a simple dleegate that returns a no-op idisposable.

Reply:
> i thought using was null tolerant anyways

This is correct.

https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZQAs6AsgBQCU+hxTUADOgG4CGATugB7oAvOgAiAbkZEAvskkl0ASREBLAM4AHAParOAIwA2AU1FCAfOgAuAC26aA7ugB2AV336JSKUA===

Reply:
Hmm, I recall getting an NRE for this somewhere due to Invoke being called on a null delegate. I'll investigate..